java/Nikiforov-Oleksandr/src/com/zoolatech/lecture7/tasks/_2/SynchronizedCalculator.java

```diff
@@ -0,0 +1,31 @@
+package com.zoolatech.lecture7.tasks._2;
+
+import java.sql.SQLOutput;
+
+public class SynchronizedCalculator extends Calculator {
+
+    public SynchronizedCalculator(int value) {
+        super(value);
+    }
+
+    @Override
+    public void add(int valueToAdd) {
+        synchronized (this) {
```

Comment:
it's better to define the whole method as synchronized - the functionality remains the same, but it requires less code